docs/data/date-pickers/fields/LifeCycleDateRangeField.tsx

```diff
@@ -19,7 +19,7 @@ export default function LifeCycleDateRangeField() {
         <SingleInputDateRangeField
           value={value}
           onChange={(newValue) => setValue(newValue)}
-          sx={{ width: 400 }}
+          sx={{ width: 300 }}
```

Comment:
Without

<img width="406" alt="Screenshot 2023-02-11 at 00 47 57" src="https://user-images.githubusercontent.com/3165635/218223093-d8c0634b-e69c-4ef9-8944-978c1b3abce5.png">

https://next.mui.com/x/react-date-pickers/fields/#when-is-onchange-called

Reply:
Maybe we could just wrap this component in `DemoContainer` and adjust it's source code with the following diff? 🤔 
```diff
diff --git a/packages/x-date-pickers/src/internals/demo/DemoContainer.tsx b/packages/x-date-pickers/src/internals/demo/DemoContainer.tsx
index 21b81841b..59f3e66ff 100644
--- a/packages/x-date-pickers/src/internals/demo/DemoContainer.tsx
+++ b/packages/x-date-pickers/src/internals/demo/DemoContainer.tsx
@@ -126,7 +126,14 @@ export function DemoContainer(props: DemoGridProps) {
   if (childrenTypes.has('UI-view')) {
     sx = null;
   } else if (childrenTypes.has('single-input-range-field')) {
-    sx = { [`& > .${textFieldClasses.root}`]: { minWidth: 400 } };
+    sx = {
+      [`& > .${textFieldClasses.root}`]: {
+        minWidth: 300,
+        md: {
+          minWidth: 400,
+        },
+      },
+    };
   } else if (childrenSupportedSections.has('date-time')) {
     sx = { [`& > .${textFieldClasses.root}`]: { minWidth: 256 } };
   } else {
```

Reply:
What's the value of having 400? Do you see a scrollbar? We could increase 300 a bit otherwise.
```suggestion
          sx={{ width: 330 }}
```

Reply:
400 is there to handle all types of `single-input-range-field` components. For `Date` or `Time` it's not a problem to have a narrower width, but for `DateTime`—we'd actually need even a bit more width. 🙈 

![Screenshot 2023-02-15 at 09 38 28](https://user-images.githubusercontent.com/4941090/218962823-a191675e-2651-4d12-b9fc-5260daaf67f8.png)

But we can leave this case as is.
I'd like to do a once-over to iron out these small inconsistencies with the `DemoContainer`. 👌